File: IKeep.Lib/Models/GenericElement.cs
using IKeep.Lib.Core;
using IKeep.Lib.Models;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace IKeep.Lib.Models
{
    public class GenericElement : Entity
    {
        public string Name { get; set; }
        public Guid ElementTypeId { get; set; }
        
        [JsonIgnore]
        public virtual ElementType ElementType { get; set; }

        public string ElementTypeName
        {
            get
            {
                return ElementType == null? "" : ElementType.Name;
            }
        }

        [JsonIgnore]
        public virtual ICollection<Element> Elements { get; set; }
        public List<Guid> ElementsIds
        {
            get
            {
                return Elements == null ? new List<Guid>() : Elements.Select(x => x.Id).ToList();
            }
        }

        [JsonIgnore]
        public virtual ICollection<GenericElementGenericTask> GenericElementGenericTasks { get; set; }
        public List<Guid> GenericElementGenericTasksIds
        {
            get
            {
                return GenericElementGenericTasks == null ? new List<Guid>() : GenericElementGenericTasks.Select(x => x.Id).ToList();
            }
        }
    }
}

File: IKeep.Web/ClientApp/src/models/GenericElement.js
import Entity from './Entity'

class GenericElement extends Entity
{
    constructor(json)
    {
        super(json);
        if(json)
        {
            this.Name = json.name;
            this.ElementTypeId = json.elementTypeId;
            this.ElementTypeName = json.elementTypeName;

            this.ElementsIds = json.elementsIds;
            /* this.GenericElementGenericChoresIds = json.genericElementGenericChoresIds;
            this.GenericChoresIds = json.genericChoresIds; */
            this.GenericElementGenericChores = json.genericElementGenericChores;
        }
        else
        {
            this.Name = "";
            this.ElementTypeId = "";
            this.ElementTypeName = "";
            this.ElementsIds = [];
            /* this.GenericElementGenericChoresIds = [];
            this.GenericChoresIds = []; */
            this.GenericElementGenericChores = [];
        }
    }
}

export default GenericElement;